### PR TITLE
fix(identity-service): resolve @ts-ignore suppressions at Credo-ts framework boundary

### PR DESCRIPTION
**Pull request by cynox-66**
## Summary

Resolves the remaining `@ts-ignore` suppressions in core `heka-identity-service` files by replacing them with typed assertions, runtime validation, and proper ambient type declarations.

This continues the type-safety cleanup started in #127 and focuses on the remaining framework/library boundary cases that required deeper fixes rather than direct `@ts-expect-error` replacements.

### 1. AnonCreds registry runtime validation

- Added a runtime guard ensuring at least one AnonCreds-capable DID method is configured
- Replaced `@ts-ignore` with a narrowed tuple assertion:
  `as [AnonCredsRegistry, ...AnonCredsRegistry[]]`
- Preserves Credo-ts type guarantees while keeping the configuration-driven registry setup

### 2. Typed AnonCreds proof access

- Replaced `@ts-ignore` in `proof.service.ts`
- Added explicit `AnonCredsProof` narrowing for `requested_proof.revealed_attrs` access

### 3. Dead field cleanup

- Removed the unused `tenantId` field and corresponding suppression in `did-registrar.service.ts`
- Preserved existing bootstrap behavior by keeping `initTenant()` execution intact

### 4. Ambient declarations for `@digitalcredentials/bitstring`

- Added local `.d.ts` declarations for the untyped dependency
- Removed related `@ts-ignore` and ESLint suppression boilerplate

## Notes

The AnonCreds registry validation now fails fast when no AnonCreds-capable DID method is configured (for example `DID_METHODS=key`).

Previously this configuration would silently create a broken agent and fail later during credential operations. The new behavior surfaces the configuration issue immediately at startup with a clear error message.

## Result

- Removes the remaining framework-boundary `@ts-ignore` suppressions in `heka-identity-service`
- Improves runtime safety for invalid DID method configurations
- Preserves existing runtime behavior for valid deployments
- Keeps Credo-ts integration points fully typed

## Summary by CodeRabbit

* **Chores**
  * Enhanced TypeScript configuration for development environment setup and improved type definitions for project dependencies.

[Review Change Stack](https://app.coderabbit.ai/change-stack/hiero-ledger/heka-identity-platform/pull/150?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: heka-identity-service/src/common/agent/agent-modules.provider.ts [hunk 1]
```diff
@@ -75,6 +75,13 @@ function getTenantModulesMap(appConfig: ConfigType<typeof AppConfig>, agencyConf
     anoncredsRegistries.push(new HederaAnonCredsRegistry())
   }
 
+  if (anoncredsRegistries.length === 0) {
```
Comment (AlexanderShenshin) [inline heka-identity-service/src/common/agent/agent-modules.provider.ts]: Is this check actually required?

It looks out of scope for type check fix while also adding a requirement to enable AnonCreds-compatible DID Method...
If there are no runtime issues with passing an empty AnonCreds registries array, I'd say we should keep the behavior as is without adding constraints.

Comment (cynox-66) [reply heka-identity-service/src/common/agent/agent-modules.provider.ts]: I removed the additional runtime validation so the PR stays scoped strictly to the type-safety cleanup and preserves the existing runtime behavior.

Instead of the tuple assertion, I switched this boundary back to a documented @ts-expect-error since the array is populated conditionally from runtime config and may legitimately be empty, while Credo-ts currently types the field as a non-empty tuple. This keeps the behavior aligned with the original implementation while still replacing the previous @ts-ignore.

While re-validating the affected files, I also cleaned up a redundant AnonCredsProof assertion in proof.service.ts that ESLint flagged as unnecessary.

Re-ran typecheck/lint on the impacted files after the changes and pushed the updated branch .

**Review by AlexanderShenshin — DISMISSED** [review]

**Comment by cynox-66** [comment]
Tracked down the CI failure ,  the ambient declaration file was included in tsconfig.src.json but not in tsconfig.test.json, so the test typecheck couldn't resolve the module declaration during transitive imports.

Added the .d.ts include to the test config and re-ran the full typecheck/lint flow locally ...

**Comment by cynox-66** [comment]
Pushed a follow-up fix for the failing typecheck related to the local @digitalcredentials/bitstring declarations.

The issue ended up being that the new .d.ts file wasn’t included in the test typecheck config, even though the source typecheck passed locally. I updated the tsconfig include patterns and re-ran the validation flow locally (check-types, lint, build, and unit tests).

Also cleaned up the ESLint issues in the declaration file while rechecking the CI flow.

Still learning some of the repo/CI edge cases here, so thanks for the patience while I worked through it carefully.

**Comment by AlexanderShenshin** [comment]
@coderabbitai review

**Review by AlexanderShenshin — APPROVED** [review]
